src/ukify/ukify.py

```diff
@@ -250,6 +251,19 @@ class Section:
     content: pathlib.Path
     tmpfile: Optional[IO] = None
     measure: bool = False
+    sections_to_show = {
+        b'.linux' : 'binary',
+        b'.initrd' : 'binary',
+        b'.splash' : 'binary',
+        b'.dtb' : 'binary',
+        b'.cmdline' : 'text',
+        b'.osrel' : 'text',
+        b'.uname' : 'text',
+        b'.pcrpkey' : 'text',
+        b'.pcrsig' : 'text',
+        b'.sbat' : 'text'
+    }
+    custom_out_file = {}
```

Comment:
That's what I have:
```python
     measure: bool = False
-    sections_to_show = {
-        b'.linux' : 'binary',
-        b'.initrd' : 'binary',
-        b'.splash' : 'binary',
-        b'.dtb' : 'binary',
-        b'.cmdline' : 'text',
-        b'.osrel' : 'text',
-        b'.uname' : 'text',
-        b'.pcrpkey' : 'text',
-        b'.pcrsig' : 'text',
-        b'.sbat' : 'text'
-    }
+
+    sections_to_show: dict[str,str] = dataclasses.field(init=False,
+        default_factory=lambda: {
+            '.linux'   : 'binary',
+            '.initrd'  : 'binary',
+            '.splash'  : 'binary',
+            '.dtb'     : 'binary',
+            '.cmdline' : 'text',
+            '.osrel'   : 'text',
+            '.uname'   : 'text',
+            '.pcrpkey' : 'text',
+            '.pcrsig'  : 'text',
+            '.sbat'    : 'text'
+        })
```

But then `Section.sections_to_show` doesn't work anymore.

Reply:
`class Section` is already a dataclass… Anyway, I was talking about `custom_out_file`, not `sections_to_show` (unless that is being modified too).

EDIT: Oh, `sections_to_show` is modified. This is very ugly.

OK. So the primary problem is that you want to modify the class in the argument parser, which is wrong.

Let's take a few steps back: you want to optionally allow the user to specify a list of sections to print. So do it like this:
1. create `DEFAULT_LIST_OF_SECTIONS_TO_PRINT = { … }` at the top level, with the dictionary you have above.
2. collect the arguments to `--read-section` in a list of tuples (`[(name, type, outfile), (name, type, outfile)]`). This list will be a field under `opts`.
3. in the function that reads sections, create `sections = DEFAULT_LIST_OF_SECTIONS_TO_PRINT | {(n,t) for (n,t,_) in opts.read_sections}`
4. use `sections` instead of `Section.sections_to_show`
5. create `files_to_write = {(n, p) for (n, _, p) in opts.read_sections}`
6. use `files_to_write` instead of `Section.custom_out_file`

I hope this helps.